boto/cloudsearch/search.py

```diff
@@ -22,9 +22,14 @@
 # IN THE SOFTWARE.
 #
 from math import ceil
-from boto.compat import json, map, six
+from boto.compat import json, six
 import requests
 
+try:
+    from collections import OrderedDict
+except ImportError:
+    from ordereddict import OrderedDict
```

Comment:
Can we move this logic into the `boto.compat` module? Then it's just `from boto.compat import OrderedDict` instead.

Also, this is introducing a new dependency. What about users on 2.6.x that don't have the OrderedDict module installed? We may need to fall back to an ordinary `dict` so that existing code in the wild does not break.